jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -876,6 +876,10 @@ private void registerRestoreEnvVariables() {
 						@SuppressWarnings("unchecked")
 						Map<String, String> newTheUnmodifiableEnvironment = (Map<String, String>) newStringEnvironment.newInstance(theEnvironment);
 
+						if (unsafe == null) {
+							initializeUnsafe();
```

Comment:
move this higher before `J9InternalCheckpointHookAPI.registerPostRestoreHook` so we dont need to do any classloading in signle thread mode